Add privacy notice to documentation index

Yashp002 opened the pull request:
Fixes #13559

Adds a brief privacy notice to the root documentation page (`docs/html/index.md`) as suggested by @ichard26.

The notice is kept short and includes:
- What data pip collects (anonymized usage data)
- What is NOT collected (personally identifiable information)
- A link to the full privacy policy

This addresses the need for transparency regarding pip's data collection practices.

ichard26 reviewed (changes requested):
Sorry, I should've been clearer. I don't think we should be implying that the pip project is directly collecting any telemetry.[^1] What pip does is send over telemetry through the User-Agent HTTP header and then it's up to the remote index to store/process that if it wishes to. In other words, I'd like something along the lines of https://github.com/pypa/pip/issues/13559#issuecomment-3247218138.

Also, the PyPA doesn't have an official privacy policy. I'm not sure where that link came from, but it returns a 404.

[^1]: Given the related issue was raised because corporate environments are sometimes put off by the linehaul service, we want to make it clear that if they use their private indices, there are zero privacy implications.

ichard26 commented:
```
The pip project does not collect any telemetry, however, pip will send non-identifying
environment information (Python version, OS, etc.) to any remote indices used, who may
choose to retain such information. Please consult PyPI's privacy policy for their
data collection and retention practices.
```

How about something like this? The wording needs work, but the general idea is there.

Unfortunately, [PyPI's privacy policy does not mention the linehaul service at all](https://policies.python.org/pypi.org/Privacy-Notice/). That seems like a glaring omission.

pfmoore commented on docs/html/index.md:
```diff
@@ -50,3 +50,10 @@ lists or chat rooms:
+
+## Privacy Notice
+
+<<<<<<< HEAD
```
You've managed to somehow leave a merge conflict marker in the file.

Yashp002 commented:
@pfmoore @ichard26 
I've updated the notice to clarify that:
- Pip itself doesn't collect or store any data
- Environment info is only transmitted to remote indices via User-Agent headers
- It's the remote index (like PyPI) that may retain the information

I've also removed the link, I seem to have followed an unverified link, my bad. Let me know if the wording further needs any improvements:)

pfmoore commented:
Thanks for your contribution, @Yashp002!

ichard26 commented on docs/html/index.md:
```diff
@@ -50,3 +50,10 @@ lists or chat rooms:
+
+## Privacy Notice
+
+The pip project does not collect any telemetry, however, pip will send
```
If someone complains that it's confusing and/or they misread it, then we can address it. Fortunately, this isn't ... like legally binding so if we get it wrong, it's not a huge deal. I'd rather see if people are confused and then deal with the ambiguity instead of debating the exact language now.
